compiler/sem/semexprs.nim

```diff
@@ -2833,8 +2833,13 @@ proc semSizeof(c: PContext, n: PNode): PNode =
   of 2:
     #restoreOldStyleType(n[1])
     n[1] = semExprWithType(c, n[1])
-    n.typ = getSysType(c.graph, n.info, tyInt)
-    result = foldSizeOf(c.config, n, n)
+    if containsGenericType(n[1].typ):
+      n[1] = c.config.newError(n[1], PAstDiag(kind: adSemTIsNotAConcreteType,
+                                              wrongType: n[1].typ.sons[0]))
```

Comment:
```suggestion
      # report the type, not the typedesc
      n[1] = c.config.newError(n[1], PAstDiag(kind: adSemTIsNotAConcreteType,
                                              wrongType: n[1].typ[0]))
```
* it's easy to overlook, so I think an explicit comment makes sense
* `typ[...]` instead of `typ.sons[...]` is generally preferred